services/gateway-api/src/circuit_breaker/fallback_handler.py

```
@@ -0,0 +1,578 @@
+    Queue for requests that need to be retried
+    """
+
+    def __init__(self, max_size: int = 1000):
+        self._queue: deque[QueuedRequest] = deque(maxlen=max_size)
+        self._processing = False
+
+    def enqueue(self, request: QueuedRequest):
+        """Add a request to the queue"""
+        if len(self._queue) >= self._queue.maxlen:
+            logger.warning(
+                f"Request queue full, dropping oldest request for {request.service_name}"
```

Ruff (0.14.8)

Logging statement uses f-string

(G004)